h2/src/test/org/h2/test/db/TestCsv.java

```diff
@@ -634,9 +644,81 @@ private void testReadEmptyNumbers2() throws Exception {
         Connection conn = DriverManager.getConnection("jdbc:h2:mem:test");
         Statement stat = conn.createStatement();
         stat.execute("CREATE TABLE TEST(TEST DECIMAL(12,2) NULL)");
-        stat.execute("INSERT INTO TEST SELECT * FROM CsvRead('" + fileName + "')");
+        stat.execute("INSERT INTO TEST SELECT * FROM CsvRead('" + fileName + "', NULL, 'quotedNulls=true')");
 
         FileUtils.delete(fileName);
     }
 
+    private void testCsvQuotedString1() throws Exception { testCsvQuotedNullStrings(false, "NULL"); }
+    private void testCsvQuotedString2() throws Exception { testCsvQuotedNullStrings(true, "NULL"); }
+    private void testCsvQuotedString3() throws Exception { testCsvQuotedNullStrings(false, ""); }
+    private void testCsvQuotedString4() throws Exception { testCsvQuotedNullStrings(true, ""); }
+    private void testCsvQuotedString5() throws Exception { testCsvQuotedNullStrings(false, "$empty"); }
+    private void testCsvQuotedString6() throws Exception { testCsvQuotedNullStrings(true, "$empty"); }
+    
+    private void testCsvQuotedNullStrings(boolean quotedStrings, String nullString) throws Exception {
+        String fileName = getBaseDir() + "/test.csv";
+        File csvFile = new File(fileName);
+        if (csvFile.exists()) {
+            csvFile.delete();
+        }
```

Comment:
H2 has own file system abstraction layer, please use it here.

Reply:
`FileUtils` used instead :thumbsup: